src/components/typography/Typography.tsx

```diff
@@ -0,0 +1,49 @@
+// src/components/Typography/Typography.tsx
+
+import React from 'react';
+import { Text, TextProps } from 'react-native';
+
+import { styles } from './Typography.styles';
+
+interface TypographyProps extends TextProps {
+  variant?: 'h1' | 'h2' | 'h3' | 'body' | 'caption';
+  color?: string;
+  fontSize?: number;
+  fontWeight?: 'normal' | 'bold' | number;
+  textAlign?: 'left' | 'right' | 'center' | 'justify';
+  children: React.ReactNode;
+}
+
+const Typography: React.FC<TypographyProps> = ({
```

Comment:
fontsize, fontweight should be not accepted as props - variant should help define it.
color - believe this should be auto set from global theme.
text align - probably should come from container
children: React.ReactNode; - Not right. remove from here and rewrite `const Typography: React.FC<PropsWithChildren<TypographyProps>> = ({ `

Highly recommend to read how Tailwind, BaseWeb and others abstract Typography components and learn from it.